packages/backend/src/utils/npm.ts

```diff
@@ -0,0 +1,127 @@
+
+    const nmLength = path.join(path.sep, "node_modules").length;
+    this.globalPath = this.globalNodeModulesPath.substring(0, this.globalNodeModulesPath.length - nmLength);
+
+    this.CHANGE_OWNER_FOR_GLOBAL = messages.change_owner_for_global(this.globalPath);
+    this.SET_DEFAULT_LOCATION = messages.set_default_location(customLocation.DEFAULT_LOCATION);
+  }
+
+  private getGenLocationParams(): string {
+    const customInstallationPath = customLocation.getPath();
+    return _.isEmpty(customInstallationPath) ? "-g" : `--prefix ${customInstallationPath}`;
+  }
+
+  public getGlobalNodeModulesPath(): string {
+    return this.globalNodeModulesPath;
+  }
+
+  public async execCommand(arg: string): Promise<any> {
+    return promisify(exec)(arg);
+  }
+
+  public getGensQueryURL(query: string, recommended: string): string {
+    return encodeURI(`${SEARCH_QUERY_PREFIX} ${query} ${recommended} ${SEARCH_QUERY_SUFFIX}`);
+  }
+
+  private async _install(genName: string): Promise<any> {
+    const locationParams = this.getGenLocationParams();
+    const command = `${NPM} install ${locationParams} ${genName}@latest`;
+    return this.execCommand(command);
+  }
+
+  public async uninstall(genName: string): Promise<any> {
+    const locationParams = this.getGenLocationParams();
+    const command = `${NPM} uninstall ${locationParams} ${genName}`;
+    return this.execCommand(command);
+  }
+
+  private async sudoExec(command: string) {
+    return new Promise((resolve, reject) => {
+      sudo.exec(command, {}, (err, script) => {
+        if (err) {
+          reject(err);
+        } else {
+          resolve(script);
+        }
+      });
+    });
+  }
+
+  private async getAccessResult(): Promise<string | undefined> {
+    // we assume that if custom path set by an user is writable
+    if (_.isEmpty(customLocation.getPath())) {
+      const isWritable = await this.isPathWritable(this.globalPath);
+      if (!isWritable) {
+        return vscode.window.showInformationMessage(
+          messages.no_write_access(this.globalPath),
+          { modal: true },
+          this.CHANGE_OWNER_FOR_GLOBAL,
+          this.SET_DEFAULT_LOCATION
+        );
+      }
+    }
+    // global path/custom path is writable
+    return HAS_ACCESS;
+  }
+
+  private isPathWritable(path: string): Promise<boolean> {
+    return promises
+      .access(path, constants.W_OK)
+      .then(() => true)
+      .catch(() => false);
+  }
+
+  public async install(genName: string) {
+    const accessResult = await this.getAccessResult();
+    if (accessResult === this.CHANGE_OWNER_FOR_GLOBAL) {
+      const changeOwnerCommand = isWin32
+        ? `icacls ${this.globalPath} /grant Users:F /Q /C /T`
+        : `chown -R $USER ${this.globalPath}`;
+      await this.sudoExec(changeOwnerCommand);
+    } else if (accessResult === this.SET_DEFAULT_LOCATION) {
+      await customLocation.setDefaultPath();
+    } else if (accessResult === CANCELED) {
```

Comment:
```suggestion
    } else if (accessResult === undefined) {
```
In case of cancel button clicked accessResult is undefined
Even with this change the generator webview is still loaded and throws errors as it is not installed.

Reply:
done